### PR TITLE
In-App Marketplace Category Banners

### PR DESCRIPTION
**Pull request by kloon**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Remove the hardcoded WC Pay and WC Services banners on the Marketplace categories pages and transform the system to use the WCCOM API and the WC Admin Rule Evaluator class for determining if a banner should be displayed on these pages.

Closes 11309-gh-Automattic/woocommerce.com

### How to test the changes in this Pull Request:

1. Check out this branch
2. Ensure your store country is set to US `WooCommerce -> Settings`
3. In WP-Admin visit `WooCommerce -> Marketplace` and verify products are still shown
4. Now navigate to the Payments category in `WooCommerce -> Marketplace` and verify the WC Payments banner is displayed above the products, if you have WC Payments installed it should not show so test with the plugin activated and deactivated.
5. Now navigate to the Shipping category in `WooCommerce -> Marketplace` and verify the WC Services banner is displayed above the products, if you have WC Services installed already it should not show so test with the plugin activated and deactivated.

### Other information:
These changes form part of the In-App Marketplace revamp.

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Tweak - Remove hardcode category banners in Settings > Marketplace and use the WooCommerce.com API instead.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by madeincosmos** [comment]
Thanks for the PR! The banner on the Payments page is looking good on all screen sizes, the Shipping one will require a fix in category names API - currently I can see it when I manually enter this URL:

https://wp.test:4443/wp-admin/admin.php?page=wc-addons&section=shipping-methods

I'm also having problems with the shipping banner image not loading correctly from https://wp.test:4443/wp-content/plugins/woocommerce/assets/images/wcs-extensions-banner-3x.png

**Comment by kloon** [comment]
Thanks Maria, [PR with API fixes ready here](11493-gh-Automattic/woocommerce.com)Once that is merged the issues should be resolved in this PR.

File: includes/admin/class-wc-admin-addons.php [hunk 1]
```diff
@@ -710,13 +760,33 @@ public static function output() {
...
+			$extension_data   = self::get_extension_data( $category, $term, $country );
+			$addons           = $extension_data->products;
+			$promotions       = ! empty( $extension_data->promotions ) ? $extension_data->promotions : array();
+		}
+
+		// We need Automattic\WooCommerce\Admin\RemoteInboxNotifications for the next part, if not remove all promotions.
+		if ( ! WC()->is_wc_admin_active() ) {
+			$promotions = array();
+		}
+		// Check for existance of promotions and evaluate out if we should show them.
```
Comment (andfinally) [inline includes/admin/class-wc-admin-addons.php]:
```suggestion
		// Check for existence of promotions and evaluate out if we should show them.
```

File: includes/admin/class-wc-admin-addons.php [hunk 2]
```diff
@@ -710,13 +760,33 @@ public static function output() {
...
+		if ( ! empty( $promotions ) ) {
+			foreach ( $promotions as $promo_id => $promotion ) {
+				$evaluator = new PromotionRuleEngine\RuleEvaluator();
+				$passed = $evaluator->evaluate( $promotion->rules );
+				if ( ! $passed ) {
+					unset( $promotions[ $promo_id ] );
+				}
+			}
+			// Tranform promotions to correct format ready for output.
```
Comment (andfinally) [inline includes/admin/class-wc-admin-addons.php]:
```suggestion
			// Transform promotions to correct format ready for output.
```

File: includes/admin/class-wc-admin-addons.php [hunk 3]
```diff
@@ -81,7 +82,7 @@ public static function build_parameter_string( $category, $term, $country ) {
 	 * @param  string $term     Search terms.
 	 * @param  string $country  Store country.
 	 *
-	 * @return array of extensions
+	 * @return object of extensions and promotions
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-addons.php]:
```suggestion
	 * @return object of extensions and promotions.
```

File: includes/admin/class-wc-admin-addons.php [hunk 4]
```diff
@@ -522,6 +523,37 @@ class="addons-img"
 		<?php
 	}
 
+	/**
+	 * Output the HTML for the promotion block
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-addons.php]:
```suggestion
	 * Output the HTML for the promotion block.
```

**Review by claudiosanches — DISMISSED** [review]
Works great!

I made some small coding standards fixes.

Just note that `get_extension_data()` is breaking compatibility now, but since it's used only as internal class I don't expect any issue from it.

**Comment by kloon** [comment]
Thanks, Claudio!

**Review by andfinally — DISMISSED** [review]
Looking good, thanks!

**Comment by andfinally** [comment]
Timur [said](https://github.com/woocommerce/woocommerce/pull/30840#issuecomment-942422984) how he got the tests working on his PR:

> Solved the problem with tests by changing hash in the composer.lock. It looks like coincidentally phpunit files in cache break running composer install.

**Review by claudiosanches — APPROVED** [review]
Looks and works great now!

**Pull request merged**